learning/tour-of-beam/frontend/lib/pages/tour/controllers/content_tree.dart

```diff
@@ -68,8 +71,12 @@ class ContentTreeController extends ChangeNotifier {
     notifyListeners();
   }
 
-  void markGroupAsCollapsed(GroupModel group) {
-    expandedIds.remove(group.id);
+  void updateExpandedIds(GroupModel group, {required bool isExpanding}) {
```

Comment:
Different behavior based on a boolean flag is an anti-pattern. This should be two separate methods: `expandGroup`, `collapseGroup`.